src/lib.rs

```diff
@@ -90,15 +93,17 @@ pub enum RibosomeError {
     RibosomeFailed(String),
     FunctionNotImplemented,
     HashNotFound,
```

Comment:
When I implemented it that way I had a conversation with a few folks, and we thought that we should operate by the principle that every language hdk should maximize the features of the language it's for to give those language users the best experience possible.  So in some other language "HashNotFound" should come back as an error, or be thrown as an exception, or come back as an nil value or what ever is normal for that language.  

This does mean that if you write zomes in different languages you have to adjust.  But that's how the choice was motivated

Reply:
This HashNotFound is only for the Rust HDK and is something different to GetEntryResult::Found/NotFound. The latter is used over the serialization/memory bridge and needs to be understood by all HDKs. The above enum RibosomeError::HashNotFound is what gets constructed locally when the received GetEntryResult (received as JSON and deserialized) is a GetEntryResult::NotFound.

I agree with @zippy that we should definitely use Rust types like Result in the Rust HDK.

But there was an open question I got from @zippy verbally about not having HashNotFound and instead stick to the Option in get_entry's signature..

Any thoughts on that?

Reply:
Sorry, I'm confused. In my comment above I spoke about code I haven't committed nor pushed :)

So right now we are using the Option and could delete HashNotFound. Or replace the option with HashNotFound error.. similar to me using RibosomeError::ValidationFailed(String) as error in commit.

I like that because it makes the calling context of our HDK functions clearer.

Reply:
I'm fine with `RibosomeError::ValidationFailed(String)`